config/crd/bases/compliance.openshift.io_customrules.yaml

```diff
@@ -0,0 +1,156 @@
+  scope: Namespaced
+  versions:
+  - name: v1alpha1
+    schema:
+      openAPIV3Schema:
+        description: CustomRule is the Schema for the customrules API
+        properties:
+          apiVersion:
+            description: |-
+              APIVersion defines the versioned schema of this representation of an object.
+              Servers should convert recognized schemas to the latest internal value, and
+              may reject unrecognized values.
+              More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources
+            type: string
+          kind:
+            description: |-
+              Kind is a string value representing the REST resource this object represents.
+              Servers may infer this from the endpoint the client submits requests to.
+              Cannot be updated.
+              In CamelCase.
+              More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+            type: string
+          metadata:
+            type: object
+          spec:
+            properties:
+              availableFixes:
+                description: The Available fixes
+                items:
+                  description: |-
+                    FixDefinition Specifies a fix or remediation
+                    that applies to a rule
+                  properties:
+                    disruption:
+                      description: |-
+                        An estimate of the potential disruption or operational
+                        degradation that this fix will impose in the target system
+                      type: string
+                    fixObject:
+                      description: an object that should bring the rule into compliance
+                      type: object
+                      x-kubernetes-embedded-resource: true
+                      x-kubernetes-preserve-unknown-fields: true
+                    platform:
+                      description: The platform that the fix applies to
+                      type: string
+                  type: object
+                nullable: true
+                type: array
+                x-kubernetes-list-type: atomic
+              checkType:
+                description: |-
+                  What type of check will this rule execute:
+                  Platform, Node or none (represented by an empty string)
+                type: string
+              description:
+                description: The description of the Rule
+                type: string
+              errorMessage:
+                description: ErrorMessage is displayed when the rule evaluation fails
+                minLength: 1
+                type: string
+              expression:
+                description: Expression is the CEL expression to evaluate
+                minLength: 1
+                type: string
+              id:
+                description: The XCCDF ID
```

Comment:
Does this have to be an `XCCDF` ID? Or does it just have to be an ID?

Reply:
I made some changes to the descriptions here, I tried to move the common part together, but it looks like we have rulePayLoad referenced in other part of the code that prevents from doing so, I wonder if we should keep the minimal changes and make the documentation clear